clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -68,6 +69,15 @@
     boolean connectionFailed(Node node);
 
     /**
+     * Check if authentication to this node has failed, based on the connection state. Authentication failures are
+     * permanent and should not be resumed in the next {@link #ready(org.apache.kafka.common.Node, long)} } call.
+     *
+     * @param node the node to check
+     * @return true iff authentication has failed
```

Comment:
Doesn't match return type of ApiException

Reply:
Thanks for catching this inconsistency. I'll fit in the next patch.